app/Http/Controllers/PostsController.php

```diff
@@ -49,12 +50,32 @@ public function store(Request $request)
     {
         $this->validate($request,[
             'title'=>'required',
-            'body'=>'required'
+            'body'=>'required',
+            'cover_image'=>'image|nullable|max:1999'
         ]);
+
+        //Handle File uploading 
+        if ($request->hasfile('cover_image')){
+            //Get file name with the extension
+            $fileNameWithExt = $request->file('cover_image')->getClientOriginalName();
+            //Get just file name
+            $fileName = pathinfo($fileNameWithExt, PATHINFO_FILENAME);
+            //Get just Ext
+            $extension = $request->file('cover_image')->getClientOriginalExtension();
+            //File name to store
+            $fileNameToStore = $fileName.'_'.time().'.'.$extension;
+            //Upload image
+            $path = $request->file('cover_image')->storeAs('public/cover_images', $fileNameToStore);
```

Comment:
storeAs creates holder by itself.

Reply:
Also, you need to link the strage to public holder. 

php artisan storage:link